### PR TITLE
Add a test for slab canaries

### PR DESCRIPTION
**Pull request by jvoisin**

### CODE REVIEW

File: test/simple-memory-corruption/canary_leak.c [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+// Check that the slab canary can't be leaked with a C-string function.
+OPTNONE int main(void) {
+    char leaked_canary[8] = {0};
+    char *p = malloc(128);
+    if (!p) {
+        return 1;
+    }
+    strncpy(leaked_canary, p + 128, sizeof(leaked_canary));
+    if (memcpy(leaked_canary, p + 128, sizeof(leaked_canary)) == 0) {
```
Comment (thestinger) [inline test/simple-memory-corruption/canary_leak.c:14]: Did you mean to use memcmp?

Comment (jvoisin) [reply test/simple-memory-corruption/canary_leak.c]: Err, absolutely.

**Comment by thestinger** [comment]
@jvoisin I think the `string_overflow.c` test already covers this. The `test_smc.py` code checks that the output is `overflow by 0 bytes`. I haven't looked at this for so long that I forgot there were per-test conditions checking output.

**Comment by jvoisin** [comment]
I think that having an explicit test for not being able to leak canaries via C-string functions is important, because `string_overflow.c` isn't super-clear :/

**Comment by thestinger** [comment]
Isn't it? It's testing that a C string function like strlen won't read past the end of the allocation with canaries enabled.

**Comment by jvoisin** [comment]
You're right :)

**Pull request closed without merging**